File: sort/BubbleSort.go
package sortPkg

import "log"

func BubbleSort(arr []int) {
	for i := range arr {
		for j := 0; j < len(arr)-1-i; j++ {
			if arr[j] > arr[j+1] {
				swap(&arr, j, j+1)
			}
		}
	}
	log.Println(arr)
}

func swap(arr *[]int, x int, y int) {
	aux := (*arr)[x]
	(*arr)[x] = (*arr)[y]
	(*arr)[y] = aux
}

File: sort/BubbleSort.ts
export default function BubbleSort(arr: number[]): void {
    for (let i = 0; i < arr.length; i++) {
        for (let j = 0; j < arr.length - 1 - i; j++) {
            if (arr[j] > arr[j + 1]) swap(arr, j, j + 1)
        }
    }
    console.log(arr)
}

function swap(arr: number[], i: number, j: number) {
    let aux = arr[i]
    arr[i] = arr[j]
    arr[j] = aux
}